cities/modeling/model_interactions.py

```diff
@@ -6,71 +7,75 @@
 import torch
 
 from cities.modeling.modeling_utils import (
-    prep_data_for_interaction_inference,
+    prep_wide_data_for_inference,
     train_interactions_model,
 )
 from cities.utils.cleaning_utils import find_repo_root
+from cities.utils.data_grabber import DataGrabber
 
 
 class InteractionsModel:
     def __init__(
         self,
         outcome_dataset,
         intervention_dataset,
-        intervention_variable,
+        intervention_variable=None,
         forward_shift=2,
         num_iterations=1500,
+        num_samples=1000,
         plot_loss=False,
     ):
         self.outcome_dataset = outcome_dataset
         self.intervention_dataset = intervention_dataset
-        self.intervention_variable = intervention_variable
         self.forward_shift = forward_shift
         self.num_iterations = num_iterations
+        self.num_samples = num_samples
         self.plot_loss = plot_loss
         self.root = find_repo_root()
 
-        data = prep_data_for_interaction_inference(
+        if intervention_variable:
+            self.intervention_variable = intervention_variable
+        else:
+            _dg = DataGrabber()
+            _dg.get_features_std_long([intervention_dataset])
+            self.intervention_variable = _dg.std_long[intervention_dataset].columns[-1]
+
+        self.data = prep_wide_data_for_inference(
             outcome_dataset=self.outcome_dataset,
             intervention_dataset=self.intervention_dataset,
-            intervention_variable=self.intervention_variable,
-            forward_shift=2,
+            forward_shift=self.forward_shift,
         )
 
-        for key, value in data.items():
-            setattr(self, key, value)
-
-        self.model = cities_model_interactions
-
-        self.model_args = (
-            self.N_t,
-            self.N_cov,
-            self.N_s,
-            self.N_u,
-            self.N_obs,
-            self.state_index_sparse,
-            self.state_index,
-            self.time_index,
-            self.unit_index,
-        )
+        self.model = model_cities_interaction
 
-        self.model_rendering = pyro.render_model(
-            self.model, model_args=self.model_args, render_distributions=True
-        )
+        self.model_args = self.data["model_args"]
 
         self.model_conditioned = pyro.condition(
             self.model,
-            data={"T": self.intervention, "Y": self.y, "X": self.covariates_sparse},
+            data={"T": self.data["t"], "Y": self.data["y"], "X": self.data["x"]},
+        )
+
+        self.model_rendering = pyro.render_model(
+            self.model, model_args=self.model_args, render_distributions=True
         )
 
     def train_interactions_model(self):
         self.guide = train_interactions_model(
```

Comment:
As alluded to elsewhere, I think putting `self.guide = train_interactions_model(...)` in the `__init__` (and having `train_interactions_model` return the guide) might be good. Same with `self.samples` in `sample_from_guide`.

Reply:
Here moving stuff to init would be less annoying for Ria as this is not used within the front end. But again, I did consider what you're suggesting. However, not doing every compute-heavy stuff at init gives the user (myself, say) better control of what step we're at, which is useful for instance for speed profiling, logging, etc, so I prefer to be able to call these separately.